### PR TITLE
Add evmzero support for PUSH0

### PR DESCRIPTION
**Pull request by LuisPH3**
Part of #465

### CODE REVIEW

**Review by facuMH — APPROVED** [review]
Looks good !! maybe add a PR description even if we had an offline discussion about it

File: cpp/vm/evmzero/interpreter.cc [hunk 1]
```diff
@@ -966,6 +967,21 @@ struct Impl<OpCode::POP> {
...
+template <>
+struct Impl<OpCode::PUSH0> {
+  constexpr static OpInfo kInfo = OpInfo{
+      .pops = 0,
+      .pushes = 1,
+      .static_gas = 2,
+      .introduced_in = EVMC_SHANGHAI,
+  };
+
+  static OpResult Run(uint256_t* top, Context&) noexcept {
```
Comment (HerbertJordan) [inline cpp/vm/evmzero/interpreter.cc:979]: The context can be skipped.
```suggestion
  static OpResult Run(uint256_t* top) noexcept {
```

File: cpp/vm/evmzero/interpreter.h [hunk 2]
```diff
@@ -54,7 +55,7 @@ struct InterpreterArgs {
   const evmc_message* message = nullptr;
   const evmc_host_interface* host_interface = nullptr;
   evmc_host_context* host_context = nullptr;
-  evmc_revision revision = EVMC_ISTANBUL;
+  evmc_revision revision = EVMC_SHANGHAI;
```
Comment (HerbertJordan) [inline cpp/vm/evmzero/interpreter.h:58]: This seems unrelated to the intent of this PR. What is the motivation for this?

File: cpp/vm/evmzero/interpreter.h [hunk 3]
```diff
@@ -111,7 +112,7 @@ struct Context {
 
   evmc::HostInterface* host = nullptr;
 
-  evmc_revision revision = EVMC_ISTANBUL;
+  evmc_revision revision = EVMC_SHANGHAI;
```
Comment (HerbertJordan) [inline cpp/vm/evmzero/interpreter.h:115]: Same here.

**Review by HerbertJordan — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments.

File: cpp/vm/evmzero/interpreter.h (hunk 2 above)
Comment (LuisPH3) [reply cpp/vm/evmzero/interpreter.h:58]: I see a number of issues with this member, I will revert it to Istanbul until I come out with a better method to provide better revision testing for evmzero (other than CT) here: #478

Some notes about the issue: 

- The default value for revision in the context struct is always overwritten, either by the effective InterpreterArgs, or by the unit test Args: This violates the single point of truth. 
- evmzero lacks of the definition for the range of supported revisions. (an equivalent to `newestSupportedRevision` in lfvm), but instead provides a baseline revision by default.  This works in conjunction with unit test running under the assumption that instructions follow Istanbul semantics unless another revision is prescribed. This unit test fails to model semantic changes that happen for some instructions between revisions. IMO it would be beneficial to explicitly test every revision support for every instruction in the unit test level as well.

File: cpp/vm/evmzero/interpreter.h (hunk 3 above)
Comment (LuisPH3) [reply cpp/vm/evmzero/interpreter.h:115]: Answered in the previous comment.

File: cpp/vm/evmzero/interpreter.h (hunk 2 above)
Comment (LuisPH3) [reply cpp/vm/evmzero/interpreter.h]: Reverted, we will come back to this issue some other time.

**Review by simonlechner — APPROVED** [review]
LGTM

**Review by facuMH — APPROVED** [review]

**Review by HerbertJordan — APPROVED** [review]
